airbyte-integrations/connectors/source-oracle-v2/src/cdk/main/kotlin/io/airbyte/cdk/command/Configuration.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.command
+
+import io.airbyte.cdk.ssh.SshConnectionOptions
+import io.airbyte.cdk.ssh.SshTunnelMethodConfiguration
+import java.time.Duration
+
+/**
+ * Interface that defines a typed connector configuration.
+ *
+ * Prefer this or its implementations over the corresponding configuration POJOs; i.e.
+ * [ConfigurationJsonObjectBase] subclasses.
+ */
+sealed interface Configuration {
+
+    val realHost: String
+    val realPort: Int
+    val sshTunnel: SshTunnelMethodConfiguration
+    val sshConnectionOptions: SshConnectionOptions
+
+    val workerConcurrency: Int
+    val workUnitSoftTimeout: Duration
+}
+
+/** Subtype of [Configuration] for sources. */
+interface SourceConfiguration : Configuration {
+
+    /** Does READ generate states of type GLOBAL? */
+    val global: Boolean
+
+    /**
+     * JDBC URL format string with placeholders for the host and port. These are dynamically
+     * assigned by SSH tunnel port forwarding, if applicable.
+     */
+    val jdbcUrlFmt: String
```

Comment:
Should this be encapsulated further - or at least named something different? We have a source that does not use JDBC (Mongo) and might have more in the future

Reply:
I'm not sure at this point. It's a great question, which I'd like to defer a bit. Perhaps we should subclass the `SourceConfiguration` further into `JdbcSourceConfiguration`. Or perhaps there's a better way.